opal/corelib/string/encoding.rb

```diff
@@ -289,7 +289,11 @@ def force_encoding(encoding)
   end
 
   def getbyte(idx)
-    @internal_encoding.getbyte(self, idx)
+    string_bytes = bytes
```

Comment:
Maybe we should memoize the value? Is there a Ruby idioms to do that? `String` is immutable right?

Reply:
That can probably be done inside `def bytes` but the we must ensure we return a different array instance each time, anyway the most common idiom is:

```rb
def bytes
  @bytes ||= each_byte.to_a
  @bytes.dup
end
```